### PR TITLE
Fix order sensitivity of sensor computation.

### PR DESCRIPTION
**Pull request by blueshiftlabs**
## Proposed change

The update() method of the sensor iterates through sensors in the order they appear in the \_resources list. Some sensors, such as the \*\_delivering and zpackages\_\* sensors, depend on the values of earlier sensors to compute their value. If the _resources list is not in exactly the right order, this causes these sensors to show the wrong value. Performing setup through the UI does not sort this list, and can cause the summary sensors (zpackages\_\*) to break.

To fix this issue, abstract the "fetch one sensor" logic into a fetch() method. That method can recursively call itself to fetch the value of any of its dependencies. The values are cached, so that a given sensor will only be computed once per update() pass, regardless of how many times that sensor is fetch()ed.

## Type of change

- [ ] Dependency upgrade
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

### CODE REVIEW

**Comment by blueshiftlabs** [comment]
@firstof9 Looks like you're the one doing most of the reviews in this repo, please take a look.

**Comment by firstof9** [comment]
Sorry for the late updates, this didn't pop into my notifications for some reason.
I'd like to test this against the 0.3.0 branch as well and see how the tests run over it before merging and release.

Thank you for the contribution.

File: custom_components/mail_and_packages/sensor.py [hunk 1]
```diff
@@ -130,72 +130,77 @@ def update(self):
...
-                    count[sensor] = update_time()
-                else:
-                    count[sensor] = get_count(
-                        account, sensor, False, self._img_out_path, self._hass
-                    )[ATTR_COUNT]
-
-                data.update(count)
+                self.fetch(account, data, sensor)
+
+            self._data = data
```
Comment (firstof9) [inline custom_components/mail_and_packages/sensor.py:135]: Updating `_data` twice

**Review by firstof9 — CHANGES_REQUESTED** [review]
Please bump the version in const.py to 0.2.3 as well.

File: custom_components/mail_and_packages/sensor.py [hunk 2]
```diff
@@ -130,72 +130,77 @@ def update(self):
...
             self._data = data
```
Comment (firstof9) [inline custom_components/mail_and_packages/sensor.py:136]: This is the 2nd update

**Review by firstof9 — CHANGES_REQUESTED** [review]

**Review by firstof9 — APPROVED** [review]

**Pull request merged**